packages/internals/src/client/getQueryEngineProtocol.ts

```diff
@@ -0,0 +1,17 @@
+import { GeneratorConfig } from '@prisma/generator-helper'
+
+export function getQueryEngineProtocol(generatorConfig?: GeneratorConfig): 'graphql' | 'json' {
+  const fromEnv = process.env.PRISMA_ENGINE_PROTOCOL
+  if (fromEnv === 'json' || fromEnv == 'graphql') {
+    return fromEnv
+  }
+
+  if (fromEnv !== undefined) {
+    throw new Error(`Invalid PRISMA_ENGINE_PROTOCOL env variable value. Expected 'graphql' or 'json', got '${fromEnv}'`)
+  }
+
+  if (generatorConfig?.previewFeatures?.includes('jsonProtocol')) {
```

Comment:
Do we have a simple functional test testing this path?

Reply:
I don't think it is possible with a current setup, unfortunately - they way it is implemented in the engine is `PRISMA_ENGINE_PROTOCOL` env var always trumps preview feature. So, we won't be able to test it on CI because it is either `PRISMA_ENGINE_PROTOCOL` is always set. I can add a unit test though.